File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
class Solution {
    static boolean[] visited;
    static ArrayList<Integer>[] adjList;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];
        adjList = new ArrayList[n];
        for(int i = 0; i < n; i++) {
            adjList[i] = new ArrayList<>();
        }
        
        for(int i = 0; i < computers.length; i++) {
            for(int j = 0; j < computers[i].length; j++) {
                if(computers[i][j] == 1) {
                    adjList[i].add(j);
                }
            }
        }
        
        for(int i = 0; i < n; i++) {
            if(!visited[i]) {
                dfs(i, 0);
                answer++;
            }
        }
    
        return answer;
    }
    
    private static int dfs(int curr, int cnt) {
        visited[curr] = true;
        for(int next : adjList[curr]) {
            if(!visited[next]) {
                dfs(next, cnt + 1);
            }
        }
        return cnt;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    let visited = Array(n).fill(false);
    
    for(let i = 0; i < n; i++) {
        if(!visited[i]) {
            dfs(i);
            answer++;
        }
    }
    
    function dfs(idx) {
        visited[idx] = true;
        
        for(let i = 0; i < n; i++) {
            if(computers[idx][i] == 1 && !visited[i]) {
                visited[i] = true;
                dfs(i);
            }
        }
    }
    return answer;
}